separate.js

```diff
@@ -0,0 +1,35 @@
+import { dropRight } from 'lodash'
+import each from './each'
+
+/**
+ * This method like join, but that returns an array
+ *
+ * @since 1.0.0
+ * @category Array
+ * @param {Array} array The array of grouped elements to process.
+ * @param {any} separator The element separator.
+ * @returns {Array} Returns the an array.
+ * @example
+ *
+ * const result = separate(['a', 'b', 'c'], '!')
+ * // => ['a', '!', 'b', '!', 'c']
+ */
+function separate(array, separator) {
+  if (!(array != null && array.length)) {
+    return []
+  }
+
+  if (separator == null) {
+    return array
+  }
+
+  const result = []
+  each(array, (element, index) => {
+    result.push(element)
+    result.push(separator)
+  })
+
+  return dropRight(result)
```

Comment:
Do you think adding (iterating) until the `lastIndex - 1` index and pushing it at the end is better?   
We could save one an extra `dropRight` execution.

Reply:
Could you write me a code sample

Reply:
Was thinking of something like the following. What do you think is the better approach?

```es6
let result = []

for (let i = 0; i < array.length - 1; i++) {
  result.push(array[i])
  result.push(separator)
}

result.push(array[array.length - 1])

return result
```

Reply:
But now that I think about it, the following would be slightly performant since we are not dynamically updating the array size.

```es6
const finalLength = array.length * 2 - 1;
const result = new Array(finalLength);

for (let i = 0; i < array.length - 1; i++) {
  const resultIndex = i * 2;
  result[resultIndex] = array[i];
  result[resultIndex + 1] = separator;
}

result[result.length - 1] = array[array.length - 1];
```